modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -179,16 +186,25 @@ public AsyncIndexBySearchAction(BulkByScrollTask task, ESLogger logger, ParentTa
         @Override
         protected ScrollableHitSource buildScrollableResultSource(BackoffPolicy backoffPolicy) {
             if (mainRequest.getRemoteInfo() != null) {
-                // NORELEASE track 500-level retries that are builtin to the client
                 RemoteInfo remoteInfo = mainRequest.getRemoteInfo();
+                HttpClientBuilder httpClient = HttpClients.custom();
                 if (remoteInfo.getUsername() != null) {
-                    // NORELEASE support auth
-                    throw new UnsupportedOperationException("Auth is unsupported");
+                    UsernamePasswordCredentials creds = new UsernamePasswordCredentials(remoteInfo.getUsername(), remoteInfo.getPassword());
+                    CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                    credentialsProvider.setCredentials(AuthScope.ANY, creds);
+                    httpClient.setDefaultCredentialsProvider(credentialsProvider);
```

Comment:
Right. I'm throwing out whatever defaults you have here which is bad. I wonder if it'd be better to expose the customization points on the builder so that you can't pass in an http client. So it is harder to throw out the defaults.